cleanup

nerdrew opened the pull request:
- make constants private
- re-introduce explicit @logger (drops support for ruby < 2.3)
- cleanup tests, refactor acceptance tests
- upgrade dev dependencies, fix rubocop lints
- add simplecov

baweaver commented on lib/zk_recipes/cache.rb:
```diff
@@ -182,35 +184,33 @@ def update_cache(path)
+        return
+      end
+
+      unless !@zk.connected?
```
Odd negation, `if @zk.connected?` may be clearer

nerdrew replied:
Good catch. Fixed.

nerdrew committed:
cleanup

- make constants private
- re-introduce explicit @logger (drops support for ruby < 2.3)
- cleanup tests, refactor acceptance tests
- upgrade dev dependencies, fix rubocop lints
- add simplecov

nerdrew commented on lib/zk_recipes/cache.rb:
```diff
@@ -6,6 +6,7 @@ class Error < StandardError; end
     class PathError < Error; end
 
     AS_NOTIFICATION = "cache.zk_recipes"
+    USE_DEFAULT = Object.new
```
This is an existing public CONSTANT, moving to the the top of the file.

nerdrew commented on lib/zk_recipes/cache.rb:
```diff
@@ -182,35 +184,33 @@ def update_cache(path)
       end
     end
 
-    %w(debug warn error).each do |m|
```
I didn't love this experiment. Move the logs back to explicit calls.

baweaver commented on spec/spec_helper.rb:
```diff
@@ -1,10 +1,27 @@
+  # Start SimpleCov as usual.
+  SimpleCov.start do
+    add_filter "/spec/"
+  end
```
Do we want to leave SimpleCov in?

nerdrew replied:
I think so... I thought it was helpful.

havenwood commented on spec/stress_spec.rb:
```diff
@@ -1,7 +1,7 @@
-RSpec.describe "stress test", zookeeper: true, proxy: true, stress: true do
-  let(:logger) { Logger.new(ENV["ZK_RECIPES_DEBUG"] ? STDERR : nil) }
+RSpec.describe "stress test", zookeeper: true, proxy: true, slow: true do
+  let(:logger) { Logger.new(ENV["ZK_RECIPES_DEBUG"].present? ? STDERR : nil) }
```
Just a nit, but if `ENV["ZK_RECIPES_DEBUG"] = "0"` it'll still be present, `"0".present? #=> true`.

I think it might be more straightforward logic to just check if the env var is set, `STDERR if ENV.key?("ZK_RECIPES_DEBUG")`. It doesn't handle the empty String case, but it almost seems better to simply rely on the var being set or not if not following env var truthy conventions.

Just an aside, but `ActiveModel::Type::Boolean.new.cast("0") #=> false`.

nerdrew replied:
Ya, I've gone back and forth on this. Gonna leave as is since it is only in tests.
